src/Microsoft.Tye.Extensions/Dapr/DaprExtension.cs

```diff
@@ -20,12 +20,67 @@ public override Task ProcessAsync(ExtensionContext context, ExtensionConfigurati
 
             if (context.Operation == ExtensionContext.OperationKind.LocalRun)
             {
+                // default placement port number
+                var placementPort = 50005;
+                var isCustomPlacementPort = false;
+
+                var existingDaprPlacementDefinition = context.Application.Services.FirstOrDefault(s =>
+                    s is ContainerServiceBuilder serviceBuilder && serviceBuilder.Image == "daprio/dapr");
+
+                // see if a placement port number has been defined at the extension level
+                if (config.Data.TryGetValue("placement-port", out var obj) && obj?.ToString() is string && int.TryParse(obj.ToString(), out var customPlacementPort))
+                {
+                    context.Output.WriteDebugLine($"Using Dapr placement service port {customPlacementPort} from 'placement-port'");
+                    placementPort = customPlacementPort;
+                    isCustomPlacementPort = true;
+                }
+
+                // check to see if a custom service definition for the placement container exists.
+                if (existingDaprPlacementDefinition == null)
+                {
+                    context.Output.WriteDebugLine("Dapr placement service not defined in Tye.yaml. Trying to create one...");
+
+                    if (!(config.Data.TryGetValue("exclude-placement-container", out obj) &&
+                          obj?.ToString() is string excludePlacementContainer && excludePlacementContainer == "true"))
+                    {
+                        context.Output.WriteDebugLine("Injecting Dapr placement service...");
+                        var daprPlacement = new ContainerServiceBuilder("placement", "daprio/dapr")
+                        {
+                            Args = "./placement",
+                            Bindings = {
+                                new BindingBuilder() {
+                                    Port = placementPort,
+                                    ContainerPort = 50005,
+                                    Protocol = "http"
+                                }
+                            }
+                        };
+                        context.Application.Services.Add(daprPlacement);
+                    }
+                    else
+                    {
+                        context.Output.WriteDebugLine("Skipping injecting Dapr placement service because 'exclude-placement-container=true'...");
+                    }
+                }
+                else
+                {
+                    // use the port defined in the custom service definition is possible
+                    var definedPlacementPort = existingDaprPlacementDefinition.Bindings.FirstOrDefault(b => b.Port.HasValue);
```

Comment:
Based on my suggestion earlier, I think if someone sets `excludePlacementService`, they must define placement-port so we know what that is.

Reply:
Yes I added this check now. It will ignore `exclude-placement-service` if not `placement-port` has been defined.